core/core.go

```
@@ -433,6 +433,22 @@ func (c *Core) SyncTargetEntropy() (*big.Int, *big.Int) {
 	}
 }
 
+// TriggerSnapSync triggers snap sync at the zone level
+func (c *Core) TriggerSnapSync(header *types.Header) {
+	if header == nil {
+		log.Global.Warn("TriggerSnapSync called with nil header")
```

This should be a fatal error. This can only happen if we make a programming error and its non-nonsensical for the node to continue operation in that case

In this case, I think its fine not to check, and let the code panic later if this ever is actually nil